crates/miden-objects/src/block/account_witness.rs

```diff
@@ -148,6 +150,14 @@ impl AccountWitness {
         SmtProof::new(self.path, leaf)
             .expect("merkle path depth should be the SMT depth by construction")
     }
+
+    /// Returns an iterator over every inner node of this witness' merkle path.
+    pub fn inner_nodes(&self) -> impl Iterator<Item = InnerNodeInfo> + '_ {
+        let leaf = self.leaf();
+        self.path()
+            .inner_nodes(leaf.index().value(), self.leaf().hash())
+            .expect("leaf index is u64 and should be less than 2^SMT_DEPTH")
```

Comment:
```suggestion
        let leaf = self.leaf();
        self.path()
            .inner_nodes(leaf.index().value(), leaf.hash())
            .expect("leaf index is u64 and should be less than 2^SMT_DEPTH")
```

Nit: I know it was my suggestion, but I think we should reuse the leaf, but it doesn't matter that much either.

Reply:
Agreed, looks better